xbmc/pvr/addons/PVRClients.cpp

```diff
@@ -1195,23 +1193,25 @@ bool CPVRClients::OpenStream(const CPVRChannel &tag, bool bIsSwitchingChannel)
   return bReturn;
 }
 
-bool CPVRClients::OpenStream(const CPVRRecording &tag)
+bool CPVRClients::OpenStream(const CPVRRecordingPtr &tag)
 {
   bool bReturn(false);
-  CloseStream();
-
-  /* try to open the recording stream on the client */
-  PVR_CLIENT client;
-  if (GetConnectedClient(tag.m_iClientId, client) &&
-      client->OpenStream(tag))
+  if (tag)
   {
-    CSingleLock lock(m_critSection);
-    m_playingClientId = tag.m_iClientId;
-    m_bIsPlayingRecording = true;
-    m_strPlayingClientName = client->GetFriendlyName();
-    bReturn = true;
-  }
+    CloseStream();
```

Comment:
This changes the logic, before `CloseStream` was called regardless of whether the tag was valid.